src/data_info/services/trash.clj

```diff
@@ -177,42 +186,42 @@
           (set-owner cm parent user)
           (recur (ft/dirname parent)))))))
 
-(defn- restore-path
+(defn- restore-paths
   [{:keys [user paths user-trash]}]
-  (irods/with-jargon-exceptions [cm]
-    (let [paths (mapv ft/rm-last-slash paths)]
-      (if (seq paths)
-        (do
-          (validators/user-exists cm user)
-          (validators/all-paths-exist cm paths)
-          (validators/all-paths-writeable cm user paths)
-
-          (let [retval (atom (hash-map))]
-            (doseq [path paths]
-              (let [fully-restored      (ft/rm-last-slash (restoration-path cm user path))
-                    restored-to-homedir (restore-to-homedir? cm user path)]
-                (log/warn "Restoring " path " to " fully-restored)
-
-                (validators/path-not-exists cm fully-restored)
-                (log/warn fully-restored " does not exist. That's good.")
-
-                (restore-parent-dirs cm user fully-restored)
-                (log/warn "Done restoring parent dirs for " fully-restored)
-
-                (validators/path-writeable cm user (ft/dirname fully-restored))
-                (log/warn fully-restored "is writeable. That's good.")
-
-                (log/warn "Moving " path " to " fully-restored)
-                (validators/path-not-exists cm fully-restored)
-
-                (log/warn fully-restored " does not exist. That's good.")
-                (move cm path fully-restored :user user :admin-users (cfg/irods-admins))
-                (log/warn "Done moving " path " to " fully-restored)
-
-                (swap! retval assoc path {:restored-path fully-restored
-                                          :partial-restore restored-to-homedir})))
-            {:restored @retval}))
-        {:restored {}}))))
+  (let [paths (mapv ft/rm-last-slash paths)]
+    (if (seq paths)
+      (irods/with-jargon-exceptions [cm]
+        (validators/user-exists cm user)
+        (validators/all-paths-exist cm paths)
+        (validators/all-paths-writeable cm user paths)
+
+        (let [retval (apply merge (mapv
+                                    (fn [path]
+                                      {path (restoration-paths cm user path)})
+                                    paths))]
+          (log/warn retval)
```

Comment:
I wasn't sure if you still wanted this to be logged as a warning, so I thought I'd mention it in case you didn't.

Reply:
Oops, I probably didn't, now that you mention it. Thanks!